File: activity/src/main/java/com.activity/ActivityParam.java
package com.activity;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.ActivityOptionsCompat;
import android.view.View;


/**
 * The type Activity param.
 */
public class ActivityParam {

    Context context;
    Class<?> uri;
    int requestCode = 0, flag = 0, enterAnim = 0, exitAnim = 0;
    boolean isAnimation = false;
    ActivityType activityType = ActivityType.START;
    Bundle bundle;
    ActivityOptionsCompat activityOptionsCompat;
    View[] sharedElements;

    public enum ActivityType {
        START,
        START_RESULT,
        FINISH,
        START_FINISH,
        START_RESULT_FINISH
    }
}

File: activity/src/main/java/com/activity/ActivityParam.kt
package com.activity

import android.content.Context
import android.os.Bundle
import android.support.v4.app.ActivityOptionsCompat
import android.view.View


/**
 * The type Activity param.
 */
class ActivityParam {

    internal var context: Context? = null
    internal var uri: Class<*>? = null
    internal var requestCode = 0
    internal var flag = 0
    internal var enterAnim = 0
    internal var exitAnim = 0
    internal var isAnimation = false
    internal var activityType = ActivityType.START
    internal var bundle = Bundle()
    internal var activityOptionsCompat: ActivityOptionsCompat? = null
    internal var sharedElements: Array<View> = arrayOf()

    enum class ActivityType {
        START,
        START_RESULT,
        FINISH,
        START_FINISH,
        START_RESULT_FINISH
    }
}
